### ISSUE TITLE
Question about RGCN example.

### ISSUE THREAD

**Issue by Derek-Wds** [dmlc/dgl#3401]
## ❓ Questions and Help

Hi, I have two small questions regarding to the implementation of Pytorch RGCN. 

1. It seems that in the `link_prediction.py` file, the `train_data` is used to build the test graph. (Reference line [here](https://github.com/dmlc/dgl/blob/master/examples/pytorch/rgcn/link_predict.py#L115).) 
2. During validation phase, it seems that `test_data` is used to validate model performance and calculate metrics such as filtered MRR and raw MRR. (Reference codes [here](https://github.com/dmlc/dgl/blob/master/examples/pytorch/rgcn/link_predict.py#L187) and [here](https://github.com/dmlc/dgl/blob/master/examples/pytorch/rgcn/utils.py#L302)). Does this suggest that we are picking the best model on the test?

I feel like these two details might be related to the filtered MRR metric computation, but are they still valid for raw MRR? Can anyone explain it a bit to me? I'm really puzzled here. Thanks in advance!

**Comment by Rhett-Ying** [comment]
we just refer to Code: https://github.com/MichSchli/RelationPrediction. could you turn to the original author for more info?

**Comment by Derek-Wds** [comment]
There is no much difference regarding to the performance on other link prediction dataset. I just wonder if it would be better to use real validation and test dataset instead.

**Comment by Rhett-Ying** [comment]
and u have already tried in this way? no much difference?

**Comment by Rhett-Ying** [comment]
so according to your experiment results obtained for now, there's no much difference between `test_graph generated from train_data(which is the way we used in DGL example)` and `test_graph generated from test_data(which is the way u wonder could be better)`? But generating test_graph from train_data is a bit unreasonable to you?

**Comment by Derek-Wds** [comment]
I haven't done experiments for comparing models' performance on the one in DGL implementation and the one uses`test_graph` generated from `test_data`. What I did is try to use the real test data (test graph generated test data) and compare its performance on link prediction dataset with the RGCN performance in some of the papers.

Those papers claim that they use DGL's implementation for running the RGCN baselines, but I'm not sure if they modify anything or they follow the exact same implementations as the example. From the results I got, it seems the results are similar and there are no much differences.

I might not explain things clearly above and sorry for the misunderstanding.
